### DISCUSSION TITLE
Trouble Connecting RustDesk Clients via Tailscale and RustDesk local server

### DISCUSSION THREAD

**Discussion by alessiozanutta** [rustdesk/rustdesk#7808] [General]
### Bug Description

Hello everyone,

I'm reaching out to seek assistance with a configuration issue I'm encountering while attempting to connect RustDesk clients via Tailscale. 
Here's the setup I'm working with:

**Server:**
NAS Terramaster (Local IP: 192.168.234.86) running Truenas SCALE Dragonfish-24.04-RC.1
RustDesk server installed on NAS (Truecharts, Name: rust-desk, App Version: 1.1.10.3, Chart Version: 1.2.4)
Configured ports: 21117\TCP, 21119\TCP, 21115\TCP, 21118\TCP, 21116\UDP, 21116\TCP (not opened in the router)
Tailscale installed and functioning

**Client 1:**
MacBook Pro #1 (Local IP: 192.168.234.33)
RustDesk client (Free version) installed
Tailscale installed and functioning

**Client 2:**
MacBook Pro #2 (outside local network)
RustDesk client (Free version) installed
Tailscale installed and functioning
Objective:
To establish a connection from MacBook #2 (outside the local network) to the desktop of MacBook #1.

### Background:
I'm unable to open any ports on my router, so I rely on Tailscale for access. The issue I'm facing is as follows:

### Problem:
While the local RustDesk server configuration seems successful, I'm unable to connect to the devices where I've installed and configured the RustDesk application. Tailscale is installed and working on all devices (regularly used for SMB protocol access).

In RustDesk, I've configured the server ID using the NAS device's IP address on the Tailscale network (in my case, 100.124.252.99). Additionally, I've enabled direct access via IP in the application settings. RustDesk shows a green icon and "ready" text at the bottom, indicating it's connected to the server. However, when attempting to establish a connection between the two MacBooks or via my iPhone, I receive the following error:

"Connection Error: connection reset by peer (os error 54). If direct connection is not possible, you can try via relay server."

Enabling connection via the relay server results in the following error:

"Connection Error: connection error via relay server: connection error to relay server: Please try again later."

Could anyone kindly suggest where the configuration mistake might lie and how to resolve it? Thank you very much.

### How to Reproduce

Connect to a client via Tailscale using a local server

### Expected Behavior

i'm receiving an error instead a connection to the client since in the app i see a green icon and a "ready" (see screenshot attached)

### Operating system(s) on local side and remote side

osx

### RustDesk Version(s) on local side and remote side

1.2.3 -> 1.2.3

### Screenshots

<img width="350" alt="Screenshot 2024-04-23 alle 15 46 23" src="https://github.com/rustdesk/rustdesk/assets/70146247/46d3ca47-bed7-4b7b-be06-392464ba10a9">
<img width="900" alt="Screenshot 2024-04-23 alle 15 39 24" src="https://github.com/rustdesk/rustdesk/assets/70146247/2d8af870-7988-4efc-ac94-ec62302641e3">
<img width="1156" alt="image" src="https://github.com/rustdesk/rustdesk/assets/70146247/05c62743-68c3-4982-8bf4-4fd0021ea733">
<img width="500" alt="image" src="https://github.com/rustdesk/rustdesk/assets/70146247/4ab9b321-6f7e-49f6-980a-903181a25e18">
<img width="500" alt="Screenshot 2024-04-23 alle 15 46 23" src="https://github.com/rustdesk/rustdesk/assets/70146247/383d95f0-8a5a-44e2-b84c-8256cb439b35">

**Comment by ky-bd** [comment]
I also tried using RustDesk over Tailscale, however I just connected to my devices directly with Tailscale's IP, and it works. Maybe the problem is with Tailscale's ACL and other firewall rules on the device, that the ports for RustDesk client and server need to be permitted. You may check your Tailscale's log to see if the connection is dropped by Tailscale.

**Comment by xlionjuan** [comment]
Actually, you won't need to build server since you're using Tailscale, you could enable direct IP connection to just connect your device with IP

Building server with your use case will decrease the performance of your connections.
